docs/guides/high-availability/deploy-aws-accelerator-stonith-lambda.adoc

```diff
@@ -0,0 +1,313 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/links.adoc" as links>
+
+<@tmpl.guide
+title="Deploy an AWS Lambda to guard against Split-Brain"
+summary="Building block for loadbalancer resilience"
+tileVisible="false" >
+
+This guide explains how to reduce the impact when split-brain scenarios occur between two sites in a multi-site deployment.
+
+== Architecture
+In the event of a network communication failure between the two sites in a Multi-AZ deployment, it is no
+longer possible for the two sites to continue to replicate session state between themselves and the two sites
+will become increasingly out-of-sync. As it's possible for subsequent Keycloak requests to be routed to different
+sites, this may lead to unexpected behaviour as previous updates will not have been applied to both sites.
+
+Typically in such scenarios a quorum is used to determine which sites are marked as online or offline, however as multi-site
+deployments only consist of two sites, this is not possible. Instead, we leverage the STONITH (Shoot The Other In The Head)
+pattern to ensure that once a split-brain has been detected, only one site can continue to serve user requests.
+
+STONITH is implemented by using https://prometheus.io/docs/alerting/latest/overview/[Prometheus Alerts] to call an AWS
+Lambda based webhook whenever one of the sites is unable to establish or connect to the other side. The triggered Lambda
+function inspects the current Global Accelerator configuration and removes the site reported to be offline.
```

Comment:
As above, first explain the concept, and then present the Prometheus and the AWS Lambda as one possible implementation (aka a blueprint).